dhis-2/dhis-web/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/user/MeController.java

```diff
@@ -255,6 +255,37 @@ public void updateCurrentUser( HttpServletRequest request, HttpServletResponse r
         nodeService.serialize( NodeUtils.createRootNode( collectionNode.getChildren().get( 0 ) ), "application/json", response.getOutputStream() );
     }
 
+    @RequestMapping( value = "/changePassword", method = RequestMethod.PUT , consumes = { "text/*", "application/*" } )
+    @ResponseStatus( HttpStatus.ACCEPTED )
+    public void changePassword( @RequestBody Map<String, String> body, HttpServletResponse response )
+            throws WebMessageException, NotAuthenticatedException
+    {
+        User currentUser = currentUserService.getCurrentUser();
+
+        if ( currentUser == null )
+        {
+            throw new NotAuthenticatedException();
+        }
+
+        String oldPassword = body.get( "oldPassword" );
+        String newPassword = body.get( "newPassword" );
+
+        if ( StringUtils.isEmpty( oldPassword ) || StringUtils.isEmpty( newPassword ) )
+        {
+            throw new WebMessageException( WebMessageUtils.conflict( "OldPassword and newPassword must be provided" ) );
+        }
+
+        boolean valid = passwordManager.matches( oldPassword, currentUser.getUserCredentials().getPassword() );
+
+        if ( !valid )
+        {
+            throw new WebMessageException( WebMessageUtils.conflict( "OldPassword is incorrect" ) );
```

Comment:
The reason we should ask for "OldPassword" is to authenticate the user, since we are doing a sensitive change. Considering that, I think it makes sense to include that fact in the error message, for example:
"Current password provided is incorrect".